cmd/proxy.go

```diff
@@ -4,106 +4,92 @@ import (
-hostname and port where it listens.'`,
+The server connects to a metadata store, currently only MLMD is supported. It supports options to customize the
+hostname and port where it listens.`,
 	RunE: runProxyServer,
 }
 
 func runProxyServer(cmd *cobra.Command, args []string) error {
-	var conn *grpc.ClientConn
+	var (
+		dsTeardownF datastore.TeardownFunc
+		wg          sync.WaitGroup
+	)
 
 	router := proxy.NewDynamicRouter()
 
 	router.SetRouter(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		http.Error(w, mlmdUnavailableMessage, http.StatusServiceUnavailable)
+		http.Error(w, datastoreUnavailableMessage, http.StatusServiceUnavailable)
 	}))
 
 	errChan := make(chan error, 1)
-	var wg sync.WaitGroup
+
 	wg.Add(2)
+
 	go func() {
 		defer close(errChan)
 		wg.Wait()
 	}()
 
-	// Start the connection to the MLMD server in a separate goroutine, so that
+	// Start the connection to the Datastore server in a separate goroutine, so that
 	// we can start the proxy server and start serving requests while we wait
 	// for the connection to be established.
 	go func() {
-		defer wg.Done()
-
-		mlmdAddr := fmt.Sprintf("%s:%d", proxyCfg.MLMDHostname, proxyCfg.MLMDPort)
-		glog.Infof("connecting to MLMD server %s..", mlmdAddr)
-		var err error
-		conn, err = grpc.NewClient(mlmdAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
-		if err != nil {
-			errChan <- fmt.Errorf("error dialing connection to mlmd server %s: %w", mlmdAddr, err)
-			return
-		}
-
-		mlmdTypeNamesConfig := mlmdtypes.NewMLMDTypeNamesConfigFromDefaults()
+		var (
+			ds  api.ModelRegistryApi
+			err error
+		)
 
-		// Backoff and retry GRPC requests to the MLMD server, until the server
-		// becomes available or the maximum number of attempts is reached.
-		for i := 0; i < maxGRPCRetryAttempts; i++ {
-			_, err = mlmdtypes.CreateMLMDTypes(conn, mlmdTypeNamesConfig)
-			if err == nil {
-				break
-			}
-
-			st, ok := status.FromError(err)
-			if !ok || st.Code() != codes.Unavailable {
-				errChan <- fmt.Errorf("error creating MLMD types: %w", err)
-				return
-			}
-
-			time.Sleep(time.Duration(i+1) * time.Second)
-		}
+		defer wg.Done()
 
-		service, err := core.NewModelRegistryService(conn, mlmdTypeNamesConfig)
+		ds, dsTeardownF, err = datastore.NewDatastore(proxyCfg.DatastoreType, proxyCfg.DatastoreHostname, proxyCfg.DatastorePort)
 		if err != nil {
-			errChan <- fmt.Errorf("error creating core service: %w", err)
-
+			errChan <- fmt.Errorf("error creating datastore: %w", err)
 			return
 		}
 
-		ModelRegistryServiceAPIService := openapi.NewModelRegistryServiceAPIService(service)
+		ModelRegistryServiceAPIService := openapi.NewModelRegistryServiceAPIService(ds)
 		ModelRegistryServiceAPIController := openapi.NewModelRegistryServiceAPIController(ModelRegistryServiceAPIService)
 
 		router.SetRouter(openapi.NewRouter(ModelRegistryServiceAPIController))
 
-		glog.Infof("connected to MLMD server")
+		glog.Infof("connected to Datastore server")
```

Comment:
Log the data store details it is connected to with full identification; same in the error above. of log at the beginning it is trying connect the datastore, here just announce the success or failure.

Reply:
Changed in https://github.com/kubeflow/model-registry/pull/425/commits/2486765c9ab30c09d2d673262096f14c220e60ce